crates/bevy_ecs/src/archetype.rs

```diff
@@ -221,6 +221,11 @@ impl Archetype {
         self.components.indices()
     }
 
+    #[inline]
+    pub fn component_ids(&self) -> &SparseArray<ComponentId, usize> {
```

Comment:
On second thought: could we make this `pub(crate)`. This seems overly public for this API.

Reply:
I disagree with this too. Ideally this _would_ return `&Bitset<ComponentId>` or something however `SparseArray` is the closest thing available. There's no harm in exposing this so I don't see why it should be private

Reply:
Agreed here. I just would rather us not expose the internals of the archetype, namely the internal indices of the backing sparse set. There's already a way to check if the archetype contains the component, so this just adds another, potentially more confusing way of doing that.